### DISCUSSION TITLE
GPS speed in car details view

### DISCUSSION THREAD

**Discussion by markoreg** [BimmerGestalt/AAIdrive#699] [Ideas]
Hi, I've just find that I'm missing the real speed info in my car which can be used to avoid tickets. As there is already the car details view and the gps data info in the apps menu, is it possible to add the gps speed info?

**Comment by hufman** [comment]
The [CDS data](https://bimmergestalt.github.io/BMWConnectedAnalysis/cds/) has a few speed points, and the speed that is shown in the new Car Details view uses [Speed Actual](https://github.com/BimmerGestalt/AAIdrive/blob/5c35333f211227ef973d980e4877fb7fbab7ab7e/app/src/main/java/me/hufman/androidautoidrive/carapp/carinfo/CarDetailedInfo.kt#L76). It looks like the GPS Speed is sometimes not provided through the API, with that page documenting a datapoint from the car that didn't provide a valid speed. In my own car, Speed Actual and Speed Displayed is the same.
How is your car behaving? Are you seeing a difference between Speed Actual and Speed Displayed, or what do you mean?

**Reply by markoreg** [reply]
Sorry for late response, somehow i didn't received the notification. 
The speeds displayed on the instrument cluster and in the car details is nearly identical. Maybe they is some minor difference (like 1-2km/h at higher speeds) but that can also be the result of some lag between these two displays.
Normally, the driving.speedActual should be the speed which car reads from the sensors, and driving.speedDisplayed the corrected value (usually like +5% than actual speed *percentage can vary*). Which can be more interesting here is maybe the navigation.GPSExtendedInfo.speed, but looking the sample provided in the document it is not clear if it is what I expect it to be and what is the unit of that data.

**Comment by markoreg** [comment]
Hi Hufman, I can see you have added the gps speesd into the gps car data. Great work!!! I've tested it for a few days and it works fine for now. I'll have to compare values with mobile gps data, just to be sure if it is as expected and will inform if i have any doubts.

**Reply by hufman** [reply]
I'm glad you like it!
